Chapter_1_Introduction.md

```diff
@@ -0,0 +1,424 @@
+
+Central resource files and libraries allow the separation of concerns, making the automation more modular and reusable across multiple suites, tests or tasks.
+
+The concepts of organizing are fundamental to working with Robot Framework and contribute to its flexibility and scalability in both test automation and RPA.
+
+
+
+
+## 1.4 Specification Styles
+
+> [!IMPORTANT]
+> LO-1.4 Recall the three specification styles of Robot Framework (K1)
+
+Specification styles define how the automation process or test cases are structured, focusing on how actions and verifications are expressed.
+These styles can be applied to all types of automation, including both testing and robotic process automation (RPA).
+While **Keyword-Driven Testing (KDT)** and **Behavior-Driven Development (BDD)** are commonly associated with testing, the principles behind these styles are adaptable to other forms of automation.
+
+Both styles can be mixed, even within the same test or task, but it is strongly recommended to have separate styles for separate purposes and not wildly mix them within the same body.
+So it would be one practical solution to define acceptance test cases that cover users expectations in *Behavior-Driven Style*, while these declarative Behavior-Driven keywords are implemented by calling imperative Keyword-Driven keywords.
+And other system level test cases, that are not covering acceptance criteria could be written as Keyword-Driven Testing.
+
+The approach of both styles is different in that way,
+that the *Behavior-Driven Style* is a **declarative** specification,
+where the script describe/declare what the system should do or how it should behave,
+while the *Keyword-Driven Style* is an **imperative** specification,
+where the script specifies what the automation should do to control the system.
+
+
+Beside these two different specification approaches how to write/formulate
+your automation script and their step sequences,
+there is also a third specification method, **Data-Driven Specification** that can be combined
+with the other two styles, to define the data that is used in the automation.
+
+
+
+### 1.4.1 Keyword-Driven Specification
+
+> [!IMPORTANT]
+> LO-1.4.1 Understand the basic concepts of Keyword-Driven Specification (K2)
+
+In **Keyword-Driven Specification**, automation steps are expressed through a sequence of mostly **imperative commands**.
+Keywords define the specific actions that must be executed in a particular order, similar to procedural programming.
+The emphasis is on the **actions performed by the automation/tester**.
+
+For example, in Robot Framework, a keyword-driven test might include steps like:
```

Comment:
Use of four spaces as a separator between keyword and arguments was lost in the final HTML rendering, while that was emphasized just in the previous section. 

```html
<code>Then the page title should be "Foundation | Robot Framework"</code>
```

Reply:
@datakurre great catch!